proto/src/server.rs

```diff
@@ -44,6 +44,10 @@ fn map_conversion_result<T>(res: Result<T, crate::ConversionError>) -> Result<T,
     res.map_err(|err| Status::invalid_argument(format!("{}", err)))
 }
 
+fn map_tokio_result<T>(res: Result<T, tokio::task::JoinError>) -> Result<T, Status> {
+    res.map_err(|err| Status::internal(format!("{}", err)))
+}
```

Comment:
It looks like this is always called along with `map_indradb_result`? As a result, this could be simplified so that one function handles both.

Reply:
nice I'll do that.

Reply:
K pushed a small refactor for that. I chose to keep two methods as one piece is used in the send method and seem logical to not copy that but just fold it all into a single method that unwraps and then maps the joinhandle and indra result (called `map_jh_indra_result`).

Reply:
https://github.com/indradb/indradb/pull/235/commits/87ab385d3edecee55448ff5b23e98fcb861259f5